src/other/ThreadedRecoilDiracSourceLoop.C

```diff
@@ -33,14 +33,23 @@ ThreadedRecoilDiracSourceLoop::addDefectToResult(const Point & p,
                                                  Real weight,
                                                  ThreadedRecoilDiracSourceLoop::DefectType type)
 {
-  // TODO: if weight != 1, we need to add a fractional # of results
+  // TODO: if weight != 1, we need to insert a weight into the DiracKernel for each
+  // point. That is currently not possible so still throw error here.
```

Comment:
That would be possible with a `std::map<Point /* location */, Real /*weight */> _weight;` data structure.